group-coordinator/src/test/java/org/apache/kafka/coordinator/group/assignor/GeneralUniformAssignmentBuilderTest.java

```diff
@@ -271,45 +268,37 @@ public void testReassignmentForTwoMembersThreeTopicsGivenUnbalancedPrevAssignmen
             mkMapOfPartitionRacks(4)
         ));
 
-        Map<String, AssignmentMemberSpec> members = new TreeMap<>();
+        Map<String, MemberSubscriptionSpecImpl> members = new TreeMap<>();
+        Map<Uuid, Set<Integer>> targetAssignment;
 
-        Map<Uuid, Set<Integer>> currentAssignmentForA = new TreeMap<>(
-            mkAssignment(
-                mkTopicAssignment(topic1Uuid, 0, 1, 2)
-            )
+        targetAssignment = mkAssignment(
+            mkTopicAssignment(topic1Uuid, 0, 1, 2)
         );
-        members.put(memberA, new AssignmentMemberSpec(
+        members.put(memberA, new MemberSubscriptionSpecImpl(
             Optional.empty(),
-            Optional.of("rack0"),
             Collections.singleton(topic1Uuid),
-            currentAssignmentForA
+            new Assignment(targetAssignment)
         ));
 
-        Map<Uuid, Set<Integer>> currentAssignmentForB = new TreeMap<>(
-            mkAssignment(
-                mkTopicAssignment(topic1Uuid, 3),
-                mkTopicAssignment(topic2Uuid, 0)
-            )
+        targetAssignment = mkAssignment(
```

Comment:
Note that we change from TreeMap to HashMap here. Does it matter?

Reply:
i guess it only matters for some tests, I changed them to tree maps there